chain/client/tests/catching_up.rs

```diff
@@ -641,4 +641,153 @@ mod tests {
+                                    );
+                                    for i in 0..16 {
+                                        send_tx(
+                                            &connectors1.write().unwrap()[i].0,
+                                            account_from.clone(),
+                                            account_to.clone(),
+                                            111,
+                                            1,
+                                            block.header.prev_hash,
+                                        );
+                                    }
+                                    *phase = OrphansPhases::HoldingBlock;
+                                }
+                            }
+                        }
+                        OrphansPhases::HoldingBlock => {
+                            // We don't allow to get any chunks from height `height_hold` - 1.
+                            // It means all blocks with higher height will be put into orphans.
+                            if let NetworkRequests::PartialEncodedChunkRequest { .. } = msg {
+                                if from_whom == "test1.2" {
+                                    return (NetworkResponses::NoResponse, false);
+                                }
+                            }
+                            if let NetworkRequests::PartialEncodedChunkMessage {
+                                account_id, ..
+                            } = msg
+                            {
+                                if account_id == "test1.2" {
+                                    return (NetworkResponses::NoResponse, false);
+                                }
+                            }
+                            if let NetworkRequests::Block { block } = msg {
+                                println!(
+                                    "HOLDING PHASE, HEIGHT = {:?}",
+                                    block.header.inner_lite.height
+                                );
+                                if block.header.inner_lite.height >= height_release {
+                                    *phase = OrphansPhases::ValidatingOrphans;
+                                }
+                            }
+                        }
+                        OrphansPhases::ValidatingOrphans => {
+                            if let NetworkRequests::Block { block } = msg {
+                                println!(
+                                    "VALIDATING PHASE, HEIGHT = {:?}",
+                                    block.header.inner_lite.height
+                                );
+                                assert!(block.header.inner_lite.height >= height_release);
+                                assert!(block.header.inner_lite.height <= height_release + 10);
+                                if block.header.inner_lite.height == height_release + 10 {
+                                    System::current().stop();
```

Comment:
Add a boolean that the below check happened at least once (the one that verifies balances).

I would actually expect it to fail without the epoch related changes I pointed out above, because "test1.2" should get stuck retrieving chunks from the next epoch (by sending the requests to wrong peers).

Reply:
Let's discuss offline. I expect it may work because of state syncing.

Reply:
I finally removed the test_catchup_orphans_over_epoch test. It fails because the epoch difference is too high between `test1.2` and others.